library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+require "yast"
+
+require "erb"
+
+Yast.import "Label"
+Yast.import "UI"
+
+module Yast2
+  # Class responsible for showing popups. It have small, but consistent API.
+  # Intended as replacement for Yast::Popup module.
+  # @note as UI is not easy to test, it is recommended after modifications to this class run
+  #   examples/popup_series_tester.sh which tests common combination of options.
+  # @note for rspec tests use popup_rspec where is helper for easier mocking that still does
+  # argument verifications.
+  class Popup
+    class << self
+      include Yast::I18n
+
+      # Number of lines for richtext: :auto to switch to richtext widget
+      LINES_THRESHOLD = 20
+
+      RICHTEXT_WIDTH = 60
+      RICHTEXT_HEIGHT = 10
+
+      # Shows popup and return which symbol of pressed button
+      # @param message [String] message to show
+      # @param details [String] hidden details that can be shown, if empty then it is nil shown
+      # @param headline [String, :error, :warning] sets popup headline. When String is passed it is shown as it is.
+      #   When empty string is passed no headline is shown.
+      #   If Symbol is passed it have to be from predefined set of symbols.
+      #   Note: Symbol it is just predefined strings, does not modify style of popup.
+      # @param timeout [Integer] how long wait till autoclose dialog. 0 means wait forever.
+      # @param buttons [Hash<Symbol, String>, Symbol] specified which buttons popup will have.
+      #   it can be hash in format button_id => button_text, showed in same order as in hash.
+      #   Beware that :details and :stop id is reserved.
```

Comment:
Wouldn't it be better to use other ids with lower chances of collision for those special buttons? Something like `:__details` or `:_stop_` or any other trick? Even those values are documented to be reserved is too easy to overlook that and introduce errors.

Reply:
makes sense for me, I will modify it.